File: 1030A_Search_Easy.py
"""
Entrada:
 - Primera línea: entero n (número de encuestados)
 - Segunda línea: cadena de longitud n compuesta por '0' y '1' ('1' = difícil)
Salida:
 - Imprime "HARD" si hay al menos un '1', en caso contrario "EASY"
Ejemplo:
Entrada:
3
101
Salida:
HARD
"""

n = int(input())
respuesta = 0
opinion = list(input().strip()) 
for i in range(len(opinion)):
    if opinion[i] == '1':
        respuesta = 1
        break
    else:
        respuesta = 0
if respuesta == 1:
    print("HARD")
else:
    print("EASY")

File: 1030A_Search_Easy.cpp
#include <iostream>
#include <vector>
using namespace std;

/*
Entrada:
 - Primera línea: entero n
 - Segunda línea: cadena de n caracteres '0' o '1' (opiniones)
Salida:
 - "HARD" si existe al menos un '1', en caso contrario "EASY"
Ejemplo:
Entrada:
3
101
Salida:
HARD
*/

int main() {
    
    int n;
    int resp = 0;
    string opinion;
    cin >> n;
    cin.ignore(); // Para ignorar el salto de línea después de leer n
    getline(cin, opinion); // Para leer la línea completa de opiniones después de leer n
    vector<char> opi(opinion.begin(), opinion.end());

    for (int i = 0; i < (int)opi.size(); i++) {
        if (opi [i] == '1') {
            resp = 1;
            break;
        } else {
            resp = 0;
        }
    }
    if (resp == 1) {
        cout << "HARD" << endl;
    } else {
        cout << "EASY" << endl;
    }
    return 0;
}